src/cpp/astar.cpp

```diff
@@ -104,13 +153,19 @@ static PyObject *astar(PyObject *self, PyObject *args) {
         float new_cost = costs[cur.idx] + weights[nbrs[i]];
         if (new_cost < costs[nbrs[i]]) {
           // estimate the cost to the goal based on legal moves
-          if (diag_ok) {
-            heuristic_cost = linf_norm(nbrs[i] / w, nbrs[i] % w,
-                                       goal    / w, goal    % w);
-          }
-          else {
-            heuristic_cost = l1_norm(nbrs[i] / w, nbrs[i] % w,
-                                     goal    / w, goal    % w);
+          // Get the heuristic method to use
+          if (heuristic_override == DEFAULT) {
+            if (diag_ok) {
+              heuristic_cost = linf_norm(nbrs[i] / w, nbrs[i] % w,
+                                         goal_i, goal_j);
+            } else {
+              heuristic_cost = l1_norm(nbrs[i] / w, nbrs[i] % w,
+                                       goal_i, goal_j);
+            }
+          } else {
```

Comment:
Please make the formatting here consistent

Reply:
sure, dont' have probably c++ ide setup.